OptimoveSDK/Sources/Classes/OptimoveConfig.swift

```diff
@@ -121,6 +155,14 @@ open class OptimoveConfigBuilder: NSObject {
 
         return self
     }
+    
+    
+    @objc @discardableResult public func enableDeepLinkingObjC(cname: String? = nil, _ objCHandler: @escaping (NSDictionary) -> Void) -> OptimoveConfigBuilder {
+         _deepLinkCname = URL(string: cname ?? "")
+         _objCdeepLinkHandler = objCHandler
```

Comment:
Hi @BBHOptimove, I'm not sure this is a complete solution because nothing currently calls this handler from inside the SDK with the deep link resolution. The current implementation in Optimove+DeepLinking.swift will call the OptimobileConfig.deepLinkHandler with a resolution when a deferred deep link is triggered. E.g.

- User visits example.lnk.click/promotion
- Universal Link triggers delegate
- Delegate calls SDK handler
- SDK looks up promotion details
- SDK calls deep link handler block with deep link resolution

I was thinking something more along these lines might work to bridge the implementation:

```swift 
@objc @discardableResult public func enableDeepLinking(cname: String? = nil, _ objCHandler: @escaping (NSDictionary) -> Void) {
    return self.enableDeepLinking(cname: cname, { resolution in
        objcHandler(resolution.asDict())
    }
}
```

We can implement some kind of internal helper on deep link resolution like `asDict()->NSDictionary` that performs the mapping to the dictionary. Then from the SDK's point of view, the config only ever holds one callback reference (the existing `_deepLinkHandler`) and the change is transparent to the SDK.

Happy to discuss further if there's something I'm missing.

Reply:
@objc @discardableResult public func enableDeepLinking(cname: String? = nil, _ objCHandler: @escaping (NSDictionary) -> Void) {
    return self.enableDeepLinking(cname: cname, { resolution in
        objcHandler(resolution.asDict())
    }
}

i dont understand where to write that code

Reply:
fixed

Reply:
Great, LGTM!